### PR TITLE
Remove semaphore submodule

### PR DESCRIPTION
**Pull request by cichaczem**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -13,6 +13,18 @@ pub mod poseidon_tree;
 pub mod protocol;
 pub mod util;
 
+#[must_use]
+pub const fn depth() -> usize {
```
Comment (gswirski) [inline src/lib.rs:17]: Can we make it `static SUPPORTED_DEPTH: usize` instead? It feels weird to use depth function at callsite.

**Review by gswirski — APPROVED** [review]
Looks great, minor comments

File: src/protocol.rs [hunk 2]
```diff
@@ -264,22 +264,22 @@ mod test {
             json,
             json!([
                 [
-                    "0x249ae469686987ee9368da60dd177a8c42891c02f5760e955e590c79d55cfab2",
-                    "0xf22e25870f49388459d388afb24dcf6ec11bb2d4def1e2ec26d6e42f373aad8"
+                    "0x2dc1c2e7730f1128093959e41f919c50dfc419fc2dca6252711d50e63ba7d68a",
```
Comment (gswirski) [inline src/protocol.rs:267]: This test will work only for depth = 21. Can we add one more when using d=16 to make sure it does the right thing?

**Review by gswirski — COMMENTED** [review]
As mentioned earlier, we only need depth = 16 for testing that `build.rs` does the right thing. Once @kustosz merges his lazy merkle tree, we should only use depth 21 and 30.

File: Cargo.toml [hunk 3]
```diff
@@ -65,6 +65,8 @@ tracing-test = "0.2"
 [build-dependencies]
 color-eyre = "0.6"
 enumset = "1.0.8"
+reqwest = { version = "0.11", features = ["blocking"] }
+tokio = { version = "1", features = ["full"] }
```
[the remark this answers is not in the document]
Comment (cichaczem) [reply Cargo.toml:69]: Turnes out `reqwest` downloads `tokio` as its dependency. Removed this line

File: build.rs [hunk 4]
```diff
@@ -27,31 +42,39 @@ fn absolute(path: &str) -> Result<PathBuf> {
...
 
-    // Build circuits if not exists
-    // TODO: This does not rebuild if the semaphore submodule is changed.
-    // NOTE: This requires npm / nodejs to be installed.
-    if !(zkey_file.exists() && wasm_file.exists()) {
-        run(&["npm", "install"])?;
-        run(&["npm", "exec", "ts-node", "./scripts/compile-circuits.ts"])?;
+    for extension in extensions {
+        let filename = ["semaphore", extension].join(".");
+        let download_url = [SEMAPHORE_DOWNLOAD_URL, depth_str, &filename].join("/");
```
[the remark this answers is not in the document]
Comment (cichaczem) [reply build.rs:69]: `format!` seems easier in this case

**Pull request merged**